tom-swifty/SwiftyOptions.cs

```diff
@@ -261,38 +264,62 @@ public void CheckForOptionErrors (ErrorHandling errors)
 							var targetOS = targets [0].ClangTargetOS ();
 							if (SwiftGluePath != null) {
 								string path = null;
-								if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "mac/XamGlue.framework");
-								} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "iphone/XamGlue.framework");
-								} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "appletv/XamGlue.framework");
-								} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "watch/XamGlue.framework");
+								if (!isLibrary) {
+									if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "mac/XamGlue.framework");
+									} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "iphone/XamGlue.framework");
+									} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "appletv/XamGlue.framework");
+									} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "watch/XamGlue.framework");
+									}
+								} else {
+									// TJ
```

Comment:
OK - I think we should assume that in the prefix in `/path/to/platform/XamGlue.framework`, `/path/to/` is what `SwiftGluePath` is set to.
The `FinalProduct` nonsense is something I put into the build of XamGlue to make it easier to spot the output of the build since it generates a lot of intermediate files.

Also, consider that you can avoid a cascade of if/else's with a static dictionary:
```
static Dictionary<string, string> targetOSToTargetDirectory = new Dictionary<string, string> {
        { "macos", "mac" }, { "ios", "iphone" }, { "tvos", "appletv" }, {"watchos", "watch")
};

/* ... */
     string targetOSPathPart;
     if (!targetOSToTargetDirectory.TryGet(targetOS, out targetOSPathPart)) {
        throw SomeError ();
     }
     path = Path.Combine (SwiftGluePath, $"{targetOSPathPart}/XamGlue.framework");